### DISCUSSION TITLE
back() and canGoBack() don't work after reloading the page on web

### DISCUSSION THREAD

**Discussion by nabilfreeman** [expo/router#894] [General]
I didn't open an issue for this as I'm sure you guys are aware and working on it in v3.

I noticed that the store behind `useRouter()` is not persistent and doesn't work after a page reload.

For now I have worked around it by using `window.navigation.back()` and `window.navigation.canGoBack` when the platform is web.

**Comment by fiuzagr** [comment]
On the web, `window.navigation` is an experimental feature and is not supported by all browsers. `window.location.origin` is the way to identify reloads. But when Expo Router loads nested routes, some internal redirects occur and `origin` is set.

I think Expo Router should identify the first route load, using `window.navigation` when available with fallback to `origin` and making it available via hook.
